Library/Homebrew/completions/bash.erb

```diff
@@ -21,30 +21,12 @@ then
   return
 fi
 
-__brewcomp_words_include() {
-  local element idx 
-  for (( idx = 1; idx < COMP_CWORD; idx++ ))
-  do
-    element=${COMP_WORDS[idx]}
-    [[ -n ${element} && ${element} == "$1" ]] && return 0
-  done
-  return 1
-}
-
 __brewcomp() {
   # break $1 on space, tab, and newline characters,
   # and turn it into a newline separated list of words
-  local list s sep=$'\n' IFS=$' \t\n'
+  local list
   local cur=${COMP_WORDS[COMP_CWORD]}
-
-  for s in $1
-  do
-    __brewcomp_words_include "${s}" || list+="${s}${sep}"
-  done
-
-  list=${list%"${sep}"}
-
-  IFS="${sep}"
+  list=$(echo "$1" | tr '[:space:]' '\n' | sort -u)
```

Comment:
Does this make performance worse? I'd think it might do as it's shelling out to these pipeline commands?

Reply:
> Does this make performance worse? I'd think it might do as it's shelling out to these pipeline commands?

Had to do a bit of testing: in absolute terms -- yes, this implementation is less performant. I timed execution of the completion command below (simulates autocompletion for `brew untap ` on my machine) and in the current implementation it takes 0.0008s, in the proposed implementation --  0.003s, and if I replace `| tr ...` with `"${1//[$'\t ']/$'\n'}"` -- 0.0025s (still 3 times slower). So, it looks like most of the slowness comes from shelling out (+ `| sort ...`).

<details><summary>command used for testing</summary>
```sh
__brewcomp ' homebrew/aliases homebrew/autoupdate homebrew/bundle homebrew/command-not-found homebrew/core homebrew/services homebrew/test-bot maxim-belkin/xorg'
```
</details>

Provided that it's less performant, uses non-Bash tools, and a subshell, I am myself not a big fan of this improvement. Let's keep what we currently have. I'll think about other ways to improve this part.